rustsec/src/repository/git/repository.rs

```diff
@@ -163,6 +200,116 @@ impl Repository {
 
     /// Path to the local checkout of a git repository
     pub fn path(&self) -> &Path {
-        self.path.as_ref()
+        // Safety: Would fail if this is a bare repo, which we aren't
+        self.repo.work_dir().unwrap()
+    }
+
+    /// Determines if the tree pointed to by `HEAD` contains the specified path
+    pub fn has_relative_path(&self, path: &Path) -> bool {
+        let lookup = || {
+            self.repo
+                .head_commit()
+                .ok()?
+                .tree()
+                .ok()?
+                .lookup_entry_by_path(path, &mut Vec::new())
+                .ok()
+                .map(|_e| true)
+        };
+
+        lookup().unwrap_or_default()
+    }
+
+    fn perform_fetch(repo: &mut gix::Repository) -> Result<(), Error> {
+        let mut config = repo.config_snapshot_mut();
+        config
+            .set_raw_value("committer", None, "name", "rustsec")
+            .map_err(|err| {
+                format_err!(ErrorKind::Repo, "failed to set `committer.name`: {}", err)
+            })?;
+        // Note we _have_ to set the email as well, but luckily gix does not actually
+        // validate if it's a proper email or not :)
+        config
+            .set_raw_value("committer", None, "email", "")
+            .map_err(|err| {
+                format_err!(ErrorKind::Repo, "failed to set `committer.email`: {}", err)
+            })?;
+
```

Comment:
Particularly in CI scenarios, this is needed as otherwise gix will fallback to the global config, or fail if there is not a committer set, when doing fetch and/or reflog updates.